src/bpffeature.h

```diff
@@ -1,43 +1,32 @@
 #pragma once
 
+#include <memory>
 #include <string>
 
 namespace bpftrace {
 
 class BPFfeature
 {
 public:
-  BPFfeature();
-  bool has_loop(void)
-  {
-    return has_loop_;
-  };
-  bool has_helper_send_signal(void)
-  {
-    return has_signal_;
-  };
-  bool has_helper_get_current_cgroup_id(void)
-  {
-    return has_get_current_cgroup_id_;
-  };
-  bool has_helper_override_return(void)
-  {
-    return has_override_return_;
-  };
-  bool instruction_limit(void)
-  {
-    return insns_limit_;
-  };
+  BPFfeature(){};
+
+  int instruction_limit();
+  bool has_loop();
+
+  bool has_helper_override_return();
+  bool has_helper_get_current_cgroup_id();
+  bool has_helper_send_signal();
+
   std::string report(void);
 
 protected:
-  bool has_loop_;
-  int insns_limit_;
+  std::unique_ptr<bool> has_loop_;
+  std::unique_ptr<int> insns_limit_;
 
   /* Helpers */
-  bool has_signal_;
-  bool has_get_current_cgroup_id_;
-  bool has_override_return_;
+  std::unique_ptr<bool> has_send_signal_;
+  std::unique_ptr<bool> has_get_current_cgroup_id_;
+  std::unique_ptr<bool> has_override_return_;
```

Comment:
> Seems like an inappropriate use of unique_ptr. I know we have our build set to c++14, but how about bumping to c++17 and using `std::optional`?

Agreed it is a bit hacky but should work ok for this purpose. Switching to c++17 means we have to have gcc >= 7, not sure how many issues that it going to cause (e.g. ubuntu xenial would be an issue)

Reply:
Agreed, won't block PR on this.

As a side note, I think you can install newer GCCs from this PPA: https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/test . GCC 7 has been out for almost 3 years now, I think it would be reasonable to depend on it be available.

Reply:
created an issue to look at c++17 (#1170 )